falcon/util/misc.py

```diff
@@ -320,34 +322,50 @@ def get_argnames(func):
     return args
 
 
-def get_http_status(status_code, default_reason='Unknown'):
-    """Gets both the http status code and description from just a code
+def get_http_status(status_code):
+    """Gets http status object from just a code.
 
     Args:
         status_code: integer or string that can be converted to an integer
         default_reason: default text to be appended to the status_code
             if the lookup does not find a result
 
     Returns:
-        str: status code e.g. "404 Not Found"
+        HTTPStatus: status object e.g. ``HTTPStatus.NOT_FOUND``
 
     Raises:
-        ValueError: the value entered could not be converted to an integer
-
+        ValueError: the value entered could not be converted to an integer, or
+        the requested code is not supported by the framework.
     """
     # sanitize inputs
     try:
         code = float(status_code)  # float can validate values like "401.1"
         code = int(code)  # converting to int removes the decimal places
-        if code < 100:
-            raise ValueError
     except ValueError:
         raise ValueError('get_http_status failed: "%s" is not a '
-                         'valid status code', status_code)
+                         'valid status code' % status_code)
 
-    # lookup the status code
+    # lookup the status object
     try:
         return getattr(status_codes, 'HTTP_' + str(code))
     except AttributeError:
-        # not found
-        return str(code) + ' ' + default_reason
+        raise ValueError('get_http_status failed: %s is not a '
+                         'supported status code' % status_code)
+
+
+def get_http_status_line(status):
```

Comment:
Suggestions for a better name are very welcome as this is a bit of a misnomer.

Technically HTTP status line includes HTTP protocol, so a full status would look like ``HTTP/1.1 404 Not Found``.